--- arch_test.neorv32-riscof.h	2022-09-19 16:18:10.716372400 +0200
+++ riscv-arch-test/riscv-test-suite/env/arch_test.h	2022-09-19 16:28:32.834930300 +0200
@@ -364,7 +364,16 @@
   
   adj_mtval:
         	csrr   t2, CSR_MCAUSE  /* code begin adjustment amount already in t3 */
-  
+
+// ----- BEGIN riscv-arch-test.mtval_ebreak patch -----
+  mtval_ebreak_patch_begin:
+          addi t4, zero, 3
+          bne  t4, t2, mtval_ebreak_patch_end
+          SREG zero, 3*REGWIDTH(t1) // mtval is zero on ebreak (mcause==3) instruction
+          j    resto_rtn
+  mtval_ebreak_patch_end:
+// ----- END riscv-arch-test.mtval_ebreak patch -----
+
           LI(t4, CODE_REL_TVAL_MSK)   /* trap#s 12, 3,1,0, -- adjust w/ code_begin */
           sll     t4, t4, t2		          /* put bit# in MSB */
           bltz    t4, sv_mtval		        /* correct adjustment is code_begin in t3 */
